test_programs/compile_failure/unsigned_integer_literal_overflow/src/main.nr

```diff
@@ -1,5 +1,5 @@
 fn main() {
-    foo(1234)
+    foo(256)
```

Comment:
I changed this to prove that it starts erroring as soon as we reach the maximum value plus one... though I guess either is fine given that we have the other Rust tests.